### DISCUSSION TITLE
fileupload extension name auto changed

### DISCUSSION THREAD

**Discussion by pslxx** [livewire/livewire#6640] [1. Help]
The.p12 automatically changes to.bin
The.crt .pen to .txt

<img width="1074" alt="image" src="https://github.com/livewire/livewire/assets/8417207/fa0504ab-71f8-428a-baba-da6132eee89e">

How do you fix it

**Comment by cododel** [comment]
Same with csv, js, css, and any text files...
All question is abandoned, im tired search the causes....

Why i cant preserve file extension, include for tmp file?!
Works only for getUploadedFileNameForStorageUsing + acceptedFileTypes((['text/javascript', 'text/plain'])
But first - filters all non js files, second filters all non text files
On uploading, in file manager - i can choose any text file, but it will be rejected by text/javascript
If ill remove text/plain - all files will be rejected by "File must be text/javascript type", but no files, event with .js extension - not has text/javascript mimetype. It all text/plain.

"text/javascript" Should be set to "accept" html attribute, on backend should be validated as text/plain and filename should be endswith .js exception!

Why no solves for this...?!
